perf(gatsby): Create one promise per queue, not one per job

pvdz opened the pull request:
Before it was creating one Promise per job and doing an `await Promise.all` on them.

Afterwards it will listen to the event the queue will emit when all events have finished and then resolves a promise.

This way, at scale, we don't have to create thousands of promises (one per page) which means that this change eliminates one (small) source of scaling problems.

vladar commented on packages/gatsby/src/query/queue.js:
```diff
@@ -100,15 +92,22 @@ const processBatch = async (queue, jobs, activity) => {
+      .on(`task_failed`, (...err) => {
+        // Note: the first arg is the path, the second the error
+        reject(err)
+        return err
```
Curious, why would we `return` from the listener?

pvdz replied:
[Looks like](https://github.com/diamondio/better-queue/blob/master/lib/worker.js#L15) it's using https://nodejs.org/api/events.html and in there I don't see a use for the return value. So I'll drop it :) Good suggestion

vladar commented on packages/gatsby/src/query/queue.js:
```diff
@@ -100,15 +92,22 @@ const processBatch = async (queue, jobs, activity) => {
+        return err
+      })
+      // Note: `drain` fires when all tasks _finish_, `empty` fires when queue is empty (but tasks are still running)
+      .on(`drain`, resolve)
```
Should we clean up listeners at some point? Maybe use `once` for both `task_failed` and `drain` instead of `on`?

pvdz replied:
This queue is created for this task and it will be gc'ed once it's released in processQueries (in query/index.js). So I don't think we need to tear the listeners down.

I'm fine with using `once` instead of `on`. Shouldn't make a difference but is cleaner.

pvdz commented:
I've dropped the return and changed to use `once` (this should also do cleanup after firing).

The queue is created for this task and as far as I can see released once it's done. It should not persist after the `drain` callback fires.
